File: 2015/day01/perl/day01p1.pl
#2025 Levi D. Smith

open(f, "../input01.txt");
while($strLine = <f>) {
#  print "$strLine\n";
  chomp($strLine);
  @chars = split(//, $strLine);
  $iValue = 0;
  foreach my $c (@chars) {
    if ($c eq '(') {
        $iValue += 1;
    } elsif ($c eq ')') {
        $iValue -= 1;
    }
  }
  print "$iValue\n";
}

File: 2015/day01/python/day01p1.py
#2025 Levi D. Smith
def main():
  with open("../input01.txt", "r") as f:
    strLine = f.readline()
    while (strLine != ''):
#      print(strLine)
      iValue = 0
      for char in strLine:
        if char == '(':
          iValue += 1
        elif char == ')':
          iValue -= 1

      print(iValue)
      strLine = f.readline()


if __name__ == "__main__":
  main()
